docker-cogstack/gate/Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM cogstacksystems/cogstack-pipeline:dev-latest
+
+RUN apt-get update && \
+	apt-get install -y curl && \
+	apt-get clean autoclean && \
+	apt-get autoremove -y && \
+	rm -rf /var/lib/apt/lists/*
+
+# GATE directories structure:
+# - core components: /gate/home/
+# - custom user apps: /gate/app/
+RUN mkdir -p /gate/
+WORKDIR /gate/
+
+# for the moment use the older GATE bundle containing all plugins and core components 
+# TODO: update to GATE 8.5+
+RUN curl -L 'https://downloads.sourceforge.net/project/gate/gate/8.4.1/gate-8.4.1-build5753-BIN.zip?r=https%3A%2F%2Fsourceforge.net%2Fprojects%2Fgate%2Ffiles%2Fgate%2F8.4.1%2Fgate-8.4.1-build5753-BIN.zip' > gate-8.4.1-build5753-BIN.zip && \
```

Comment:
Should be fine, but one nice option might be to leverage using Docker-ized Gate to parallelize gate by running several containers, simultaneously would probably be sensible to break the gate application out into it's own micro-service.

Reply:
That's a good idea. Currently, there are actually two ways of getting the NLP part with GATE running CogStack pipeline:
1. using `gate` profile, which uses GATE Embedded to extract the annotations using GATE API  (this example)
2. using `webservice` profile, which will use an external webservice (here: any dockerized GATE microservice; BioYodie apparently has been also used that way) with exposed REST endpoint

The 2. option gives more flexibility than 1. (e.g. can configure the parallelisation of GATE as you say), yet may degrade a bit the performance. However, we may integrate any external application this way. I would like to cover in the next step the latter example with BioYodie. 

We can later come back to this topic, check and decide which option would be the recommended one to use.

Reply:
I wouldn't worry too much about the overhead of the webservice, I don't think that'll be the overall bottleneck.

Also been thinking about how we might deploy optional microservices in RADAR take a look at this too might be useful for ways to compose things https://docs.docker.com/compose/extends/

Reply:
Thanks! Extending configs with optional microservices will definitely become handy -- btw. in most examples I'm using the default microservices stack (postgres, elasticsearch, kibana) and this can help with reducing the code duplication in config files.